hillelDjango3/schema.py

```diff
@@ -132,10 +144,40 @@ def mutate(self, info, id, title=None, price=None, summary=None, is_18_plus=None
             return UpdateProductMutation(product=None, error=str(e))
 
 
+class CreateOrderMutation(graphene.Mutation):
+    class Arguments:
+        user_id = graphene.ID(required=True)
+        products = graphene.List(OrderProductInput, required=True)
+
+    order = graphene.Field(OrderObjectType)
+    user_errors = graphene.List(graphene.String)
+    error = graphene.String()
+
+    def mutate(self, info, user_id, products):
+        try:
+            user = User.objects.get(pk=user_id)
+        except User.DoesNotExist:
+            return CreateOrderMutation(order=None, user_errors=['User does not exist.'])
+
+        if not products:
+            return CreateOrderMutation(order=None, user_errors=['At least one product is required.'])
+
+        try:
+            order = Order.objects.create(user=user)
+            for product in products:
+                OrderProduct.objects.create(order=order, product=product.product, quantity=product.quantity)
```

Comment:
in this case order will be already created however, it will be incomplete and lost. Please wrap your mutate method into transaction